### PR TITLE
dpdk: added VERIFY-DPDK-VPP TC

### PR DESCRIPTION
**Pull request by ader1990**
Added VPP DPDK test case.

VPP repo: https://github.com/FDio/vpp.git
VPP is a software switch implementation, created initially by CISCO
and open sourced.

Only VPP 18.10 or higher is supported for Azure,
as the MLX PCI ids from Azure are not supported in previous versions.

VPP 19.02 or higher supports DPDK 18.11 or 19.02.
VPP 18.10 supports DPDK 18.05 or 18.11.

VPP tag v18.10 does not work on Ubuntu 18.04 (JAVA code bug)

VERIFY-DPDK-VPP TC deploys an Azure VM with 3 accelerated nics, builds
and installs VPP with DPDK Mellanox CX4 support for Azure, starts vpp
process and checks if the MLX interfaces show up in the vpp interfaces
as failsafe devices.

Supported OSses: RHEL and Ubuntu. Tested on Ubuntu 18.04 and RHEL 7.5. On Ubuntu 19.04, the vpp compilation fails due to linux syscall incompatibility.

The test fails if the fortyGigabyte interfaces do not show up.

### CODE REVIEW

File: Testscripts/Windows/VERIFY-DPDK-VPP.ps1 [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
```
Comment (juhlee-microsoft) [inline Testscripts/Windows/VERIFY-DPDK-VPP.ps1:2]: It is nice to put some brief information about VPP. After reviewing code lines, it is enough to understand. The extra description can expedite the reviewer's comprehension speed.

File: Testscripts/Linux/build_test_vpp.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+function build_test_vpp () {
+	SetTestStateRunning
+	LogMsg "Configuring ${1} ${DISTRO_NAME} ${DISTRO_VERSION} for VPP test..."
+	packages=(git)
+	package_manager=""
+	package_manager_install_flags=""
+	package_type=""
+	case "${DISTRO_NAME}" in
+		oracle|rhel|centos)
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/build_test_vpp.sh:32]: In PS, it filters out others except Redhat and Ubuntu. This line should be only 'rhel'. The line 46 is 'ubuntu'.

**Review by juhlee-microsoft — DISMISSED** [review]
Please add description - I think this is VPP build and test if VPP can find MLX hardware.

**Comment by ader1990** [comment]
Execution tested on RHEL 7.5

```
Test Run On           : 05/27/2019 13:17:28
ARM Image Under Test  : RedHat : RHEL : 7.5 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:29

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 DPDK                 VERIFY-DPDK-VPP                                                                   PASS                25.06

```

File: Testscripts/Linux/build_test_vpp.sh (hunk 2 above)
Comment (ader1990) [reply Testscripts/Linux/build_test_vpp.sh:32]: the PS rules do not apply, as it s bash.

File: Testscripts/Windows/VERIFY-DPDK-VPP.ps1 (hunk 1 above)
Comment (ader1990) [reply Testscripts/Windows/VERIFY-DPDK-VPP.ps1:2]: I ve put the info

**Comment by ader1990** [comment]
Discussing internally if the VPP should test the failsafe and/or native DPDK pmd driver. Currently, the native DPDK MLX driver is not working on RHEL and the test will fail.

**Review by chvalean — APPROVED** [review]

**Comment by ader1990** [comment]
RHEL test execution, which currently fails (current expected failure, as the kernel seems to not have all the required patches):

```
Test Run On           : 05/27/2019 17:05:32
ARM Image Under Test  : RedHat : RHEL : 7.5 : latest
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:35

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 DPDK                 VERIFY-DPDK-VPP                                                                   FAIL                 25.7
```

**Review by chvalean — APPROVED** [review]

**Pull request merged**